### ISSUE TITLE
Kotlin suspend functions generate response schema of Object rather than specific type (still)

### ISSUE THREAD

**Issue by csbiggar** [micronaut-projects/micronaut-openapi#255]
(Re-reporting #234 as I can still recreate the issue)

When using micronaut-openapi to generate a swagger spec for a kotlin suspend function, the response type is not being recognised, instead it's being generated as Object always.

Please see this fork for an example - details of how to reproduce the error are in the readme:
https://github.com/csbiggar/creating-your-first-micronaut-app-kotlin

This is the same example project as last time, [upgraded] (https://github.com/csbiggar/creating-your-first-micronaut-app-kotlin/commit/31ee2e5de4f89c2993d925f48fd1d6a85d51e887) to release [1.5.1](https://github.com/micronaut-projects/micronaut-openapi/releases/tag/v1.5.1) which includes the fix for #234 - the issue is still reproducable

Please could you take another look? Thanks

**Comment by croudet** [comment]
Working fine on my side:
```yaml
openapi: 3.0.1
info:
  title: Hello World
  description: Demonstrate suspend functions not recognising return type
  contact:
    name: name
    url: url
    email: email
  license:
    name: Apache 2.0
    url: http://foo.bar
  version: "0.0"
paths:
  /hello/{name}:
    get:
      operationId: hello
      parameters:
      - name: name
        in: path
        required: true
        schema:
          type: string
      responses:
        default:
          description: hello default response
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/Name'
  /hello/suspend/{name}:
    get:
      operationId: helloSuspend
      parameters:
      - name: name
        in: path
        required: true
        schema:
          type: string
      responses:
        default:
          description: helloSuspend default response
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/Name'
components:
  schemas:
    Name:
      type: object
      properties:
        firstName:
          type: string
        lastName:
          type: string
```

**Comment by csbiggar** [comment]
Hmm! That is very weird. It doesn't work for me :

```  hello-micronaut git:(master) date 
Tue  9 Jun 20:21:19 BST 2020
➜  hello-micronaut git:(master) git remote -v 
origin	git@github.com:csbiggar/creating-your-first-micronaut-app-kotlin.git (fetch)
origin	git@github.com:csbiggar/creating-your-first-micronaut-app-kotlin.git (push)
➜  hello-micronaut git:(master) git pull 
Already up-to-date.
➜  hello-micronaut git:(master) ./gradlew build run  

> Task :complete:run
20:21:46.516 [main] INFO  io.micronaut.runtime.Micronaut - Startup completed in 600ms. Server Running: http://localhost:8080
```

![image](https://user-images.githubusercontent.com/14541644/84190372-bbbc9e00-aa8e-11ea-81e1-f3e029aa3970.png)

**Comment by csbiggar** [comment]
What else could be different :thinking:  ? 

I'm running on 
```➜  hello-micronaut git:(master) java --version
openjdk 11.0.6 2020-01-14
OpenJDK Runtime Environment AdoptOpenJDK (build 11.0.6+10)
OpenJDK 64-Bit Server VM AdoptOpenJDK (build 11.0.6+10, mixed mode)
```

Ubuntu 18.04

It would be pretty surprising if that was making a difference though? Unless it's something that flaps ?  I've consistently got Object every time but maybe I've just been (un)lucky

**Comment by croudet** [comment]
Check on the file system: `less ./complete/build/tmp/kapt3/classes/main/META-INF/swagger/hello-world-0.0.yml`

**Comment by markjfisher** [comment]
I've just downloaded the project and compiled it, the file is wrong for me too:

    ...
    /hello/suspend/{name}:
      get:
        operationId: helloSuspend
        parameters:
          - name: name
            in: path
            required: true
            schema:
              type: string
        responses:
          default:
            description: helloSuspend default response
            content:
              application/json:
                schema:
                  $ref: '#/components/schemas/Object'

**Comment by markjfisher** [comment]
I've just upgrade the project to Micronaut 1.3.6 and it is consistently working now.

**Comment by croudet** [comment]
Yes it works with 1.3.6

**Comment by csbiggar** [comment]
> Check on the file system

Yeah, it's definitely wrong there too, sorry I should have included that 

> upgrade the project to Micronaut 1.3.6 
This fixes it for me too! 

>Yes it works with 1.3.6
Brill, will upgrade our project. Bit confused as to why you didn't see the error with 1.3.5 and we did, but :woman_shrugging:

**Comment by croudet** [comment]
It is because my 1.3.5 jar is patched with the fix from 1.3.6, sorry.

I will propose a PR to fix this on micronaut-openapi side.

**Comment by alvarosanchez** [comment]
Fixed by #256
